Layout fixes; consolidating typesetting into SCSS; refreshed Visualization Guide

beep opened the pull request:
Hey folks! Got a big ol’ PR here, which I’d love to treat as a _discussion starter_ more than anything. There’s a lot that’s changed here, and I’d love to get as many thoughts as possible before (and if) we fold it in.

Basically, I had two goals for opening this up:

1. To do a visual QA/refinement pass on our v2 redesign;
1. To consolidate type styling and layout control in our SCSS. (And—hopefully!—retire [reflexbox](https://rebassjs.org/reflexbox/) and [typography.js](https://kyleamathews.github.io/typography.js/) as a result.)

This touched a fair bit of the codebase, so I’d appreciate any thoughts folks have on this. And I’d _really_ appreciate some design review as well, as I’ve tried tightening up a few things throughout the site. (Nothing major, mind—mainly some more consistent margins and heading treatments—but still! Feedback! I love it!)

Additionally, per some discussion below, we now have a new `featured` layout for our Visualization Guide. Thanks so much to @jasonsantamaria and @julialedur for the design feedback!

Code-wise, there’s one additional item I snuck in: I renamed `<GetInvolved />` to `<ListArrow />` in https://github.com/COVID19Tracking/website/pull/638/commits/3d14a6ac019a8f939865afccc765c5a80c9d8120. This is _totally_ out of scope for this PR, but _also_ please treat it as a proposal! It seemed like a more generic name for this component might be helpful since we’re starting to use it in more places. But as with everything else above, I’d be more than happy to discuss/revert this if others would prefer.

Thanks for your feedback, all!

kevee commented:
I am *very much* pro getting rid of things like typography and moving layout like flexboxes back into their component modules. Thank you so much for doing this.

vijaysharwar commented:
I'm also in favor of removing Typography in favor of SCSS only. @goleary @jeremiak we should test `/dashboard` out with these changes. I imagine it will require some local adjustments.

Also @beep any reason not to delete `/src/utilities/typography.js`?

beep commented:
> I am very much pro getting rid of things like typography and moving layout like flexboxes back into their component modules. Thank you so much for doing this.

Oh, that’s _wonderful_ to hear, @kevee—thank you! I realize this is going to touch a bunch of tiny parts of our codebase, so I’d love to make sure this looks good to you and @schwartzadev.

My goal is to get this opened up for review early next week, but I’ll keep y’all apprised of my progress.

> I'm also in favor of removing Typography in favor of SCSS only.

That’s _great_ to hear, @vijaysharwar! This is definitely still very much a draft PR; I’ll keep you folks posted on how I get on. If y’all want to coordinate changes in the dashboard, I’d love your feedback once this is opened up for review.

> Also @beep any reason not to delete /src/utilities/typography.js?

@vijaysharwar None! I’m still working on this PR, so I’ve been keeping the old infrastructure in place to test my work as I go. I’ll be sure that’s deleted once this PR opens up for review. Thank you for noting that!

kevee commented on gatsby-config.js:
```diff
@@ -121,5 +115,11 @@ module.exports = {
+    {
+      resolve: `gatsby-transformer-remark`,
+      options: {
+        plugins: [`gatsby-remark-smartypants`],
```
The new plugin should just be added to the prior entry for gatsby-transformer-remark entry in line 86

beep replied:
@kevee Eep, I totally missed that—thank you! How’s da5d778206bb30ae54bf9ba934b9532b9bde232e look to you?

kevee commented on src/components/pages/homepage/visualizations.js:
```diff
@@ -8,32 +7,32 @@ import wsjGraph from '../../../images/homepage-visualizations/wsj.png'
           sourceLink="https://www.nytimes.com/interactive/2020/03/17/us/coronavirus-testing-data.html"
           sourceTitle="U.S. Lags in Coronavirus Testing After Slow Response to Outbreak"
           publicationName="The New York Times"
+          itemClass={visualizationsStyles.visualization}
```
You should be able to import the module.scss file into the visualization.js file so you don't have to pass styles as props.

Or we can just put the `<Visualization> component into this file... it's not used anywhere else but here.

beep replied:
@kevee Oh got it, that’s good feedback—thanks, Kevin. Does aeb1ba6fba2e7d02c391c181c200c6155cc7915e look better?

beep commented:
Opening this up for review! Updated [the intro comment](https://github.com/COVID19Tracking/website/pull/638) with a bit more context for this PR, if that’s at all helpful.

(Also: as I [mentioned in Slack](https://covid-tracking.slack.com/archives/C010N48RY73/p1587501733253900), I updated the snapshots referenced by jest in 5bee88e166e2fcae798c9bc0f25105f3fe163bc9 with `npm run test:coverage -- -u`; if someone could make sure I didn’t donk anything up, I’d deeply appreciate it.) (And if I _did_ donk them up, do let me know and I’ll revert/revisit.

vijaysharwar commented:
@beep testing `/dashboard` on the latest here, and noticing many minor layout issues. I think it's a single root cause: removing `Typography` has also removed a top level setting of `box-sizing: border-box`. Generally, I usually set this on all my projects, as it's easier to calculate widths of containers inclusive of their internal paddings, especially when working with percentage layouts/responsiveness.

Adding this to `global.scss` does the trick, and seems to resolve the layout issues I'm seeing:

```
html {
  box-sizing: border-box;
}

*,
*:before,
*:after {
  box-sizing: inherit;
}
```

Otherwise, there may be some minor spacing issues that I think @goleary or @jeremiak can assess and fix quicker (since I don't have access to this repo).

Also, I think we're launching dashboard V1 very soon, at a new URL path. I think this PR should be put on hold until that happens, try to keep major changes to a minimum.

toddparker commented:
This is great @beep. On small screens I'm seeing some rough edges in the header and menu. Not sure if all these are intentional but some things feel a wee bit more crowded.

![image](https://user-images.githubusercontent.com/359929/79914817-e32ead00-83f3-11ea-9157-d71925695498.png)

beep commented:
@vijaysharwar Ach, I’m embarrassed to say I completely neglected to review the dashboard while I was working. I’ll prioritize that tomorrow—thank you!

@toddparker _Great_ catch, thank you. Looks like I dropped a `flex-direction` in moving the header off of reflexbox; I’ll sort that out in the morning.

superbiche commented:
@beep both @vijaysharwar and @toddparker issues are fixed (and it's *indeed* a bit less crowded), there was also a `key` warning in `list-arrow.js` which I were able to fix

Also found a `validateDOMNesting(...): <col> cannot appear as a child of <table>` in `summary-table.js` which would need to be investigated first (I'd have to wrap my mind around all the project).

It's https://github.com/COVID19Tracking/website/pull/679 if someone can check the table col issue

superbiche commented:
Also if someone is able to reproduce the radiusless bottom-left menu visible on https://github.com/COVID19Tracking/website/pull/638#issuecomment-617419035 that would be great, it happens in none of the browsers I can test on (Only FF/Chrome, Linux user)

vijaysharwar commented:
@superbiche @beep Thanks for checking on the issues I raised. But I'm still not seeing `box-sizing: border-box;` applied globally, and it's affecting the layout. cc @goleary @jeremiak @julialedur.

In particular, the state multiple charts are breaking, because we have percentage padding applied to each item. Without `box-sizing: border-box;`, the padding is *added* to the percentage width (rather than being included), so items no longer fit correctly in a flex box row.

Compare screen shot from this branch, pink area is incorrect dead space:

![Screen Shot 2020-04-22 at 11 50 37 AM](https://user-images.githubusercontent.com/63169602/80004008-ce075c00-848f-11ea-84de-c5e175f029b7.png)

With current live page at https://covidtracking.com/about-data/visualization-guide:

![Screen Shot 2020-04-22 at 11 50 46 AM](https://user-images.githubusercontent.com/63169602/80004059-df506880-848f-11ea-8ed1-a04436c3cf26.png)

Other issues:

1. Needs to be brought up-to-date with `master`. `/dashboard` has been moved to `/about-data/visualization-guide`
2. Play/stop button in the US map is broken (this is resolved by fixing `box-sizing: border-box;`):

![Screen Shot 2020-04-22 at 11 54 11 AM](https://user-images.githubusercontent.com/63169602/80004310-29394e80-8490-11ea-95cc-fa6616ed57ad.png)

3. The content column or perhaps `<p>` has changed, which makes the page layout look really odd (this is _not_ fixed by `box-sizing: border-box;`):

![Screen Shot 2020-04-22 at 11 56 27 AM](https://user-images.githubusercontent.com/63169602/80004670-9baa2e80-8490-11ea-9a09-3f3141d16eae.png)

`master` and live site:

![Screen Shot 2020-04-22 at 11 56 44 AM](https://user-images.githubusercontent.com/63169602/80004694-a5339680-8490-11ea-850a-e07957feb160.png)

schwartzadev commented:
Hey all! I'm going to try to make order out af chaos here and make a checklist for remaining issues here:
- [ ] Definitively apply global bounding box ([comment](https://github.com/COVID19Tracking/website/pull/638#issuecomment-617871028))
- [ ] Fix the header order ([comment](https://github.com/COVID19Tracking/website/pull/638#issuecomment-617422698)), #679 
> @toddparker _Great_ catch, thank you. Looks like I dropped a `flex-direction` in moving the header off of reflexbox; I’ll sort that out in the morning.

New PRs:
- [ ] Fix the borderless nav issue (if this is an issue?) ([comment](https://github.com/COVID19Tracking/website/pull/638#issuecomment-617864976))


Please add new items to this comment -- hopefully it'll keep this organized so we can get it out the door soon!

Thank you all for the great work on this 🎉

vijaysharwar commented:
@beep @superbiche in case it's helpful, submitted a PR to this `beep/type-fixes` branch with the global `box-sizing` fix. https://github.com/COVID19Tracking/website/pull/682

superbiche commented:
Thanks @vijaysharwar for fixing the misunderstanding and @schwartzadev for the checklist, that helps a lot. 
I have some stuff to deal with at work first but should be able to get back working on it in an hour.

schwartzadev commented:
Re: the content spacing ![](https://user-images.githubusercontent.com/63169602/80004670-9baa2e80-8490-11ea-9a09-3f3141d16eae.png)

What type of behavior do we want? I image two options:
1. The "narrow" layout, where all layout content is aligned to the left on large viewports (including headers, which is _not_ the case above)
2. The full-width layout, where content fills the entire horizontal space

superbiche commented:
@schwartzadev I really like the full-width option: 
![image](https://user-images.githubusercontent.com/2478146/80011200-d3829780-84cb-11ea-8642-21e104bee595.png)

But that means we need to define a limit so the lines stay readable

julialedur commented:
maybe you're aware of this, but just flagging that it looks like the path to the visualization guide in the [preview link](https://deploy-preview-638--upbeat-lovelace-3e9fff.netlify.app/) is breaking @beep

superbiche commented:
Narrowed option: 

![image](https://user-images.githubusercontent.com/2478146/80011551-4f7cdf80-84cc-11ea-87db-dc9267ea5c31.png)

julialedur commented:
i also prefer the full width option! otherwise we'll end up with a LOT of white unutilized space

superbiche commented:
I can work on the full-width option tonight (on my TZ so that's the next couple hours) and the Fix the borderless nav issue.

Someone with better knowledge of the project would be better suited for the URL path item

vijaysharwar commented:
@superbiche here's another PR to this `beep/type-fixes` branch to update path for `/about-data/visualization-guide`. Also resolves the merge conflict in `src/pages/contact.js`, which should get a thorough review.

https://github.com/COVID19Tracking/website/pull/684

beep commented:
👋 Hey folks, thanks to everyone who dove in over the last few hours! I’m just getting re-situated, and sifting through all of this excellent feedback. (And a few PRs into this branch!)

A couple notes from me:

- I’d already addressed a few of these issues locally; I’ll get these pushed up, and then look into reviewing any open PRs.
- On the layout discussion!
  - Design-wise, I’d recommend we _not_ present full-width lines of text to our readers. ~1100px is a pretty long line length, and can be pretty uncomfortable to read—especially for longer passages of text.
  - But! I _definitely_ agree that the graphs should occupy the full width that’s available to them.
  - *This is all to say:* I’d like to work on a compromise layout for [the visualization guide](http://localhost:8000/about-data/visualization-guide), one that’s sort of a compromise between the “wide” and “narrow” options. I’ll review it in Slack with @julialedur and @jeremiak before bringing it back here for review.

I’m going to take this PR back to draft mode for now, as I think the timing of the Dashboard launch changes its scope a bit. I’ll reopen when it’s ready for QA & review!

Thanks again, all! I really appreciate it.

beep commented:
Okay! Round two! After some _wonderful_ discussion with @jasonsantamaria and @julialedur, we now have a new `featured` layout in use on our gorgeous Visualization Guide:

https://deploy-preview-638--upbeat-lovelace-3e9fff.netlify.app/about-data/visualization-guide

Reopening this PR for additional review; if anyone has any thoughts, I’d love to hear them!

julialedur commented:
looks awesome, @beep ! 🎉 
my only suggestion is to make the lines around the death charts full-width, for consistency sake.
they're the only narrow ones that don't run across the page: 

<img width="1279" alt="Screen Shot 2020-04-22 at 5 51 17 PM" src="https://user-images.githubusercontent.com/39416238/80037716-235c6100-84c2-11ea-8059-27a9fe8f020a.png">

beep commented:
>  my only suggestion is to make the lines around the death charts full-width, for consistency sake.

@julialedur Oh good call! I had to _slightly_ tweak the layout of the death charts, as they were constrained to that narrower width. (That’s why the borders weren’t going full-width.) I’ve updated the layout slightly in 6fea6181cdec45a3c1167f848501629faa427c38; how’s it look to you now?

jasonsantamaria commented:
This looks great! All of the design tweaks really gave this a nice flow. 🙌

julialedur commented:
looks great to me! 👍

schwartzadev commented:
Hey All!

This is looking _spectacular_ but I wanted to flag the following:
![image](https://user-images.githubusercontent.com/18607205/80052697-8eb12d80-84d8-11ea-8f64-0e785dea43ae.png)

The width on the `DetailText` for tables is being reduced to `max-width: 40em`. I'm not sure if this is intended behavior, but I wanted to make it clear that this was happening.

Also, the computed `DetailText` `font-size` increased from `14.4px` to `16px`.

(The numbers in the table are outdated; this is only on my development environment.)

beep commented:
@schwartzadev Thanks so much for asking! That _is_ intended behavior, on both fronts. The line length on those notes needed a tiny bit of reining in, especially on some of the longer passages.

This is intended as a patch job for now, pending the bigger table redesign that @toddparker’s heading up.

schwartzadev commented:
Excellent, thanks @beep!
